packages/flutter/lib/src/material/slider.dart

```diff
@@ -535,6 +541,30 @@ class _RenderSlider extends RenderBox {
 
   bool get isDiscrete => divisions != null && divisions > 0;
 
+  @override
+  void attach(PipelineOwner owner) {
+    super.attach(owner);
+    _reaction = new CurvedAnimation(
+      parent: _state.reactionController,
+      curve: Curves.fastOutSlowIn,
+    )..addListener(markNeedsPaint);
+    _state.enableController.value = isInteractive ? 1.0 : 0.0;
+    _enableAnimation = new CurvedAnimation(
+      parent: _state.enableController,
+      curve: Curves.easeInOut,
+    )..addListener(markNeedsPaint);
+    _state.positionController.value = _value;
+    _state.positionController.addListener(markNeedsPaint);
+  }
```

Comment:
the problem with this approach is that if you move the widget tree using GlobalKey reparenting in the middle of an animation, it'll reset the animations.

Reply:
the alternative approach is to create the CurvedAnimation objects in the constructor, and set the controllers' values in the constructor (though, that's weird, you should probably do that in the State, strictly speaking, since the State owns the controller), and in attach you _only_ addListener.

Reply:
OK, will do.  I did wonder if setting the value here was OK.